bench/bm_buffer.cpp

```diff
@@ -0,0 +1,153 @@
+#include "benchmark.hpp"
+
+#include <barrier>
+#include <boost/ut.hpp>
+#include <chrono>
+#include <iostream>
+#include <thread>
+
+#include <fmt/format.h>
+
+#include <buffer.hpp> // new buffer header interface
+#include <buffer_skeleton.hpp>
+#include <circular_buffer.hpp>
+
+#include <iostream>
+
+using namespace gr;
+
+#if defined __has_include
+#if __has_include(<pthread.h>) && __has_include(<sched.h>)
```

Comment:
maybe we need to include a guard here for WASM/emscripten since the `pthread_setaffinity_np(..)` and other affinity tools are not supported on the web-browsers. See for example [here](https://www.reddit.com/r/WebAssembly/comments/em18dk/emscripten_pthread_undefined_symbol_and_thread/#t1_fdmq5gr) and [here](https://github.com/emscripten-core/emscripten/issues/7365).